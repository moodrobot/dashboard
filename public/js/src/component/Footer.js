import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class Footer extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const T = this.props.translate;
    const language = this.props.currentLanguage;
    return (
      <div className="footer">
        <div className="NavigationHeader">
          <div className="width-1440 mx-auto row list-unstyled footer-group">
            <div className="col-sm-6 col-md-3">
              <p>
                <span className="f_title">{T["header-hardware-developers"]}</span>
                <li>
                  <a href="https://www.intel.com/content/www/us/en/design/resource-design-center.html">{T["resource-and-design-center"]}</a>
                </li>
                <li>
                  <a href="http://www.intel.com/buy/">{T["shop-intel"]}</a>
                </li>
                <li>
                  <a href="https://firmware.intel.com/">{T["firmware"]}</a>
                </li>
              </p>
            </div>
            <div className="col-sm-6 col-md-3">
              <p>
                <span className="f_title">{T["header-open-source"]}</span>
                <li>
                  <a href="https://www.01.org/">{T["01-org"]}</a>
                </li>
                <li>
                  <a href="https://www.clearlinux.org/">{T["clear-linux-project"]}</a>
                </li>
                <li>
                  <a href="https://www.zephyrproject.org/">{T["zephyr-project"]}</a>
                </li>
              </p>
            </div>
            <div className="col-sm-6 col-md-3">
              <p>
                <span className="f_title">{T["header-manage-your-tools"]}</span>
                <li>
                  <a href="https://downloadcenter.intel.com/">{T["download-center"]}</a>
                </li>
                <li>
                  <a href="http://www.intel.com/supporttickets">{T["online-serivce-center"]}</a>
                </li>
                <li>
                  <a href="https://registrationcenter.intel.com/">{T["registration-center"]}</a>
                </li>
              </p>
            </div>
            <div className="col-sm-6 col-md-3">
              <p>
                <span className="f_title">{T["header-stay-up-to-date"]}</span>
                <li>
                  <a href={"/" + language + "/forum"}>{T["forums"]}</a>
                </li>
                <li>
                  <a href={"/" + language + "/recent-updates"}>{T["recent-updates"]}</a>
                </li>
                <li>
                  <a href="https://www.youtube.com/subscription_center?add_us75er=intelswnetwork">{T["subscribe-to-our-youtube-channel"]}</a>
                </li>
                <li>
                  <a href={"/" + language + "/newsletter"}>{T["newsletter-archives"]}</a>
                </li>
              </p>
            </div>
          </div>
        </div>
        <div className="SectionHeader f_terms">
          <div className="col-md-12">
            <div className="row text-nowrap">
              <div className="col-auto">
                <a  href="http://www.intel.com/content/www/us/en/legal/terms-of-use.html">{T["terms-of-use"]}</a>
              </div>
              <div className="col-auto">
                <a href="http://www.intel.com/content/www/us/en/legal/trademarks.html">{T["trademarks"]}</a>
              </div>
              <div className="col-auto">
                <a href="http://www.intel.com/content/www/us/en/privacy/intel-online-privacy-notice-summary.html">{T["privacy"]}</a>
              </div>
              <div className="col-auto">
                <a href="http://www.intel.com/content/www/us/en/privacy/intel-cookie-notice.html">{T["cookies"]}</a>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Footer.propTypes = {
  translate: PropTypes.object.isRequired,
  currentLanguage : PropTypes.string.isRequired
};

export default Footer;
