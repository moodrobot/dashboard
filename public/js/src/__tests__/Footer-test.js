import React from "react";
import { shallow } from "enzyme";
import Footer from "../component/Footer";
import i18n from "../i18n-for-testing"; // initialized i18next instance for testing

describe("<Footer />", () => {
  it("expect to match snapshot", () => {
    const wrapper = shallow(
      <Footer
        translate={i18n.t("footer")}
        currentLanguage="en-US"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
