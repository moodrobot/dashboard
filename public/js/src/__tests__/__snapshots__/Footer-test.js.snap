// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> expect to match snapshot 1`] = `
<div
  className="footer"
>
  <div
    className="NavigationHeader"
  >
    <div
      className="width-1440 mx-auto row list-unstyled footer-group"
    >
      <div
        className="col-sm-6 col-md-3"
      >
        <p>
          <span
            className="f_title"
          >
            Hardware Developers
          </span>
          <li>
            <a
              href="https://www.intel.com/content/www/us/en/design/resource-design-center.html"
            >
              Resource and Design Center
            </a>
          </li>
          <li>
            <a
              href="http://www.intel.com/buy/"
            >
              Shop Intel
            </a>
          </li>
          <li>
            <a
              href="https://firmware.intel.com/"
            >
              Firmware
            </a>
          </li>
        </p>
      </div>
      <div
        className="col-sm-6 col-md-3"
      >
        <p>
          <span
            className="f_title"
          >
            Open Source
          </span>
          <li>
            <a
              href="https://www.01.org/"
            >
              01.org
            </a>
          </li>
          <li>
            <a
              href="https://www.clearlinux.org/"
            >
              Clear Linux* Project
            </a>
          </li>
          <li>
            <a
              href="https://www.zephyrproject.org/"
            >
              Zephyr Project
            </a>
          </li>
        </p>
      </div>
      <div
        className="col-sm-6 col-md-3"
      >
        <p>
          <span
            className="f_title"
          >
            Manage Your Tools
          </span>
          <li>
            <a
              href="https://downloadcenter.intel.com/"
            >
              Download Center
            </a>
          </li>
          <li>
            <a
              href="http://www.intel.com/supporttickets"
            >
              Online Service Center
            </a>
          </li>
          <li>
            <a
              href="https://registrationcenter.intel.com/"
            >
              Registration Center
            </a>
          </li>
        </p>
      </div>
      <div
        className="col-sm-6 col-md-3"
      >
        <p>
          <span
            className="f_title"
          >
            Stay Up-to-Date
          </span>
          <li>
            <a
              href="/en-US/forum"
            >
              Forums
            </a>
          </li>
          <li>
            <a
              href="/en-US/recent-updates"
            >
              Recent Updates
            </a>
          </li>
          <li>
            <a
              href="https://www.youtube.com/subscription_center?add_us75er=intelswnetwork"
            >
              Subscribe to Our YouTube* Channel
            </a>
          </li>
          <li>
            <a
              href="/en-US/newsletter"
            >
              Newsletter Archives
            </a>
          </li>
        </p>
      </div>
    </div>
  </div>
  <div
    className="SectionHeader f_terms"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row text-nowrap"
      >
        <div
          className="col-auto"
        >
          <a
            href="http://www.intel.com/content/www/us/en/legal/terms-of-use.html"
          >
            Terms of Use
          </a>
        </div>
        <div
          className="col-auto"
        >
          <a
            href="http://www.intel.com/content/www/us/en/legal/trademarks.html"
          >
            *Trademarks
          </a>
        </div>
        <div
          className="col-auto"
        >
          <a
            href="http://www.intel.com/content/www/us/en/privacy/intel-online-privacy-notice-summary.html"
          >
            Privacy
          </a>
        </div>
        <div
          className="col-auto"
        >
          <a
            href="http://www.intel.com/content/www/us/en/privacy/intel-cookie-notice.html"
          >
            Cookies
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
